modules/steam/automount.nix

```diff
@@ -0,0 +1,108 @@
+{lib, config, pkgs, ...}: let
+  inherit (lib) mkIf mkMerge mkOption types mdDoc mkEnableOption;
+  inherit (lib.strings) concatMapStringsSep;
+  inherit (pkgs) systemd util-linux procps steam bash writeShellScriptBin ;
+  cfg = config.jovian.steam;
+  mount-script = writeShellScriptBin "mount-script" ''
+#Steam Deck Mount External Drive by scawp
+#License: DBAD: https://github.com/scawp/Steam-Deck.Mount-External-Drive/blob/main/LICENSE.md
+#Source: https://github.com/scawp/Steam-Deck.Mount-External-Drive
+# Use at own Risk!
+
+     urlencode()
+     {
+# From https://gist.github.com/HazCod/da9ec610c3d50ebff7dd5e7cac76de05
+        [ -z "$1" ] || echo -n "$@" | ${util-linux}/bin/hexdump -v -e '/1 "%02x"' | sed 's/\(..\)/%\1/g'
+     }
+     mount_drive()
+     {
+        label="$(${util-linux}/bin/lsblk -noLABEL $1)"
+        if [ -z "$label" ]; then
+          label="$(${util-linux}/bin/lsblk -noUUID $1)"
+        fi
+        mkdir -p "/mnt/$label"
+        chown ${cfg.user} -R "/mnt/$label"
+        ${util-linux}/bin/mount "$1" "/mnt/$label"
+        sleep 5
+        mount_point="$(${util-linux}/bin/lsblk -noMOUNTPOINT $1)"
+        if [ -z "$mount_point" ]; then
+          echo "Failed to mount "$1" at /mnt/$label"
+        else
+          if [ -d "$mount_point/SteamLibrary" ]; then
+            library="$mount_point/SteamLibrary"
+            url=$(urlencode "''${library}")
+            if ${procps}/bin/pgrep -x "steam" > /dev/null; then
+              ${systemd}/bin/systemd-run -M 1000@ --user --collect --wait \
+              ${bash}/bin/sh -c "${steam}/bin/steam steam://addlibraryfolder/''${url}"
+            fi
+          else
+            echo "no steam library on drive"
+          fi
+        fi
+     }
+     umount_drive()
+     {
+        label="$(${util-linux}/bin/lsblk -noLABEL $1)"
+        if [ -z "$label" ]; then
+          label="$(${util-linux}/bin/lsblk -noUUID $1)"
+        fi
+        mount_point="$(${util-linux}/bin/lsblk -noMOUNTPOINT $1)"
+        if [ -z "$mount_point" ]; then
+          echo "Failed to mount "$1" at /mnt/$label"
+        else
+          if [ -d "$mount_point/SteamLibrary" ]; then
+            library="$mount_point/SteamLibrary"
+            url=$(urlencode "''${library}")
+            if ${procps}/bin/pgrep -x "steam" > /dev/null; then
+              ${systemd}/bin/systemd-run -M 1000@ --user --collect --wait \
+                ${bash}/bin/sh -c "${steam}/bin/steam steam://removelibraryfolder/''${url}"
```

Comment:
if drive is removed while updating seem to just keep downloading, thought this would fix but doesnt